### PR TITLE
Refactor and document JIT State Logic

### PR DESCRIPTION
**Pull request by dsouzai**
Closes https://github.com/eclipse-openj9/openj9/issues/12508
Closes https://github.com/eclipse-openj9/openj9/issues/12507

### CODE REVIEW

**Comment by dsouzai** [comment]
@mpirvu could you review when you get a chance; not an urgent PR though.

File: runtime/compiler/env/J9PersistentInfo.hpp [hunk 1]
```diff
@@ -45,6 +45,42 @@ class TR_PersistentClassLoaderTable;
...
+ *        UNDEFINED_STATE: Uninitialized.
+ *
+ *             IDLE_STATE: The JVM isn't doing any work. This is when
+ *                         the application threads are doing no or
+ *                         minimal work.
+ *
+ *          STARTUP_STATE: The application is starting up; this includes
+ *                         the classloading phase. The heuristics in
+ *                         this state target ensuring the application
+ *                         starts up as quickly as possible.
```
Comment (fjeremic) [inline runtime/compiler/env/J9PersistentInfo.hpp:62]: Doxygen won't pick these up unless the documentation is right above each enum declaration.

Comment (dsouzai) [reply runtime/compiler/env/J9PersistentInfo.hpp:62]: Ah really; ok I'll look at what it should look like for enums.

**Comment by dsouzai** [comment]
@mpirvu I broke up the refactoring into smaller commits to make it easier to review. Could you take a look when you get some time?

**Pull request closed without merging**